compiler/noirc_driver/src/lib.rs

```diff
@@ -42,8 +42,12 @@ pub struct CompileOptions {
     pub print_acir: bool,
 
     /// Treat all warnings as errors
-    #[arg(long)]
+    #[arg(long, conflicts_with = "silence_warnings")]
```

Comment:
good thinking to add the conflicts_with flag since we don't want deny_warnings and silence_warnings to both be true at the same time